KarhooSDK/Network/HttpClient/TokenRefreshingHttpClient/Interactor/KarhooRefreshTokenInteractor.swift

```diff
@@ -28,15 +28,29 @@ final class KarhooRefreshTokenInteractor: RefreshTokenInteractor {
     }
 
     func refreshToken(completion: @escaping (Result<Bool>) -> Void) {
-        self.callback = completion
+        callback = completion
 
         guard tokenNeedsRefreshing() else {
             completion(Result.success(result: false))
             return
         }
 
         guard let refreshToken = dataStore.getCurrentCredentials()?.refreshToken else {
-            completion(Result.failure(error: RefreshTokenError.noRefreshToken))
+            Karhoo.configuration.requestNewAuthenticationCredentials { [weak self] newCredentials in
+                guard let self = self else { return }
```

Comment:
let's call the completion with an error here as well. I know it normally shouldn't get here, but just in case. let's not have loose ends where we end up waiting for a completion that is never triggered